src/components/common/Map.js

```diff
@@ -0,0 +1,129 @@
+import React, { PropTypes } from 'react';
+import { GoogleMapLoader, GoogleMap, Marker } from 'react-google-maps';
+import { getMyPosition } from '../../utils/MapHelper';
+
+class Map extends React.Component {
+  constructor(props, context) {
+    super(props, context);
+
+    this.onMapClick = this.onMapClick.bind(this);
+    this.success = this.success.bind(this);
+    this.error = this.error.bind(this);
+    this.getOpts = this.getOpts.bind(this);
+    this.handleMapMounted = this.handleMapMounted.bind(this);
+    this.handleCenterChanged = this.handleCenterChanged.bind(this);
+
+    const myPosCoords = JSON.parse(localStorage.getItem('myPositionCoords'));
+    if (!myPosCoords) {
+      getMyPosition(this.success, this.error);
+      this.state = {
+        locationCenter: this.props.center
+      };
+    }else {
+      this.state = {
+        locationCenter: { position: myPosCoords }
+      };
+    }
+  }
+
+  handleMapMounted(map) {
+    this._map = map;
+  }
+
+  handleCenterChanged() {
+    const mapMovement = this._map.getCenter();
+    let nextCenter = targetSelected || {
+      lat: parseFloat(mapMovement.lat().toFixed(6)),
+      lng: parseFloat(mapMovement.lng().toFixed(6))
+    };
+
+    let newState = Object.assign({}, this.state);
+    newState.locationCenter.lat = nextCenter.lat;
+    newState.locationCenter.lng = nextCenter.lng;
+    if (newState.locationCenter.lat === this.state.locationCenter.lat &&
+      newState.locationCenter.lng === this.state.locationCenter.lng &&
+      targetSelected === null) {
+      // Notice: Check nextCenter equality here,
+      // or it will fire center_changed event infinitely
+      return;
+    }
+    this.setState(newState);
+  }
+
+  getOpts(venue) {
+    let icon = '';
+    if (venue.topic !== null) {
+      icon = venue.topic.icon; //|| getTopicIcon(venue.topic, this.props.topicsList);
+    }
+
+    let opts = {};
+    if (icon !== '') {
+      opts.icon = {
+        url: icon,
+        origin: {
+          x: 0,
+          y: 0
+        },
+        anchor: {
+          x: 16,
+          y: 14
+        },
+        scaledSize: {
+          height: 28,
+          width: 28
+        }
+      };
+    }
+    return opts;
+  }
+
+  success(pos) {
+    const crd = pos.coords;
+    let newState = Object.assign({}, this.state);
+    newState.locationCenter.lat = parseFloat(crd.latitude.toFixed(6));
+    newState.locationCenter.lng = parseFloat(crd.longitude.toFixed(6));
+    localStorage.setItem('myPositionCoords', JSON.stringify(newState.locationCenter));
+    this.setState(newState);
+  }
+
+  error(err) {
+    console.warn(`ERROR( ${err.code} ):  ${err.message}`);
+  }
+
+  onMapClick(e) {
+    let lat = parseFloat(e.latLng.lat().toFixed(6));
+    let lng = parseFloat(e.latLng.lng().toFixed(6));
+    this.props.updateTargetInfo("lat", lat);
```

Comment:
Can't you convert this function take a JS object instead of running it three consecutive times? You might be forcing three different renders each time a prop is being updated. Something like:

`this.props.updateTargetInfo({ lat, lng, isVisible });`